File: nk/47266/a/a.cpp
#include <bits/stdc++.h>
using namespace std;

template <typename T>
inline void scan(T &n)
{
    n = 0; int f = 1; char ch = getchar();
    while (!isdigit(ch)) {if (ch == '-') f = -1; ch = getchar();}
    while (isdigit(ch))  {n = n * 10 + ch - '0'; ch = getchar();}
    if (f == -1) n = -n;
}

template <typename T>
void print(T n)
{
    if (n < 0) putchar('-'), n = -n;
    if (n > 9) print(n / 10);
    putchar(n % 10 + '0');
}

template <typename T>
void print(T n, char ch)
{
    print(n), putchar(ch);
}

typedef long long ll;
typedef unsigned long long ull;

const int M = (int)1e5;
const int N = (int)1e5;
const int inf = 0x3f3f3f3f;
const ll linf = 0x3f3f3f3f3f3f3f3f;
const int mod = (int)1e9 + 7;

void work()
{
    int n; scan(n);
    vector<int> v(n);
    for (auto &x: v) scan(x);
    int a, k, b, m; scan(a), scan(k), scan(b), scan(m);
    for (int i = 1; i <= m; ++i) {
        for (auto &x: v) {
            if (x > k) x = b;
            if (i < m)
                x += a;
        }
    }
    for (int i = 0; i < n; ++i) printf("%d%c", v[i], " \n"[i == n - 1]);
}

int main()
{
    // ios::sync_with_stdio(false);
    // cin.tie(0), cout.tie(0);
    int T = 1; scan(T);
    for (int ca = 1; ca <= T; ++ca) {
        // printf(
        work();
    }
    // cerr << 1.0 * clock() / CLOCKS_PER_SEC << "\n";
    return 0;
}

File: nk/47266/a/a.go
package main

import (
	"fmt"
)

const M int = 1e5

var h [M + 5]int

func work() {
	var n int
	fmt.Scan(&n)
	for i := 1; i <= n; i++ {
		fmt.Scan(&h[i])
	}
	var a, b, k, m int
	fmt.Scan(&a, &k, &b, &m)
	for i := 1; i <= m; i++ {
		for j := 1; j <= n; j++ {
			if h[j] > k {
				h[j] = b
			}
			if i < m {
				h[j] += a
			}
		}
		fmt.Println(h[n])
	}
	for i := 1; i <= n; i++ {
		fmt.Print(h[i])
		if i < n {
			fmt.Printf(" ")
		} else {
			fmt.Printf("\n")
		}
	}
}

func main() {
	var T int
	fmt.Scan(&T);
	for ca := 1; ca <= T; ca++ {
		work()
	}
}